### ISSUE TITLE
Bar chart and color combination breaks the script

### ISSUE THREAD

**Issue by vivekratnavel** [uwdata/vega-lite#537]
The following spec breaks the script and throws a lot of errors in the console.

```
{
  "marktype": "bar",
  "encoding": {
    "x": {"name": "name","type": "N"},
    "y": {"name": "Miles_per_Gallon","type": "Q"},
    "color": {"name": "Cylinders","type": "O"}
  },
  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
  "data": {"url": "data/cars.json"}
}
```

```
TypeError: Cannot read property 'push' of undefined
Uncaught TypeError: Cannot read property 'marktype' of null
```

**Comment by vivekratnavel** [comment]
The same happens with chart type area too.

```
{
  "marktype": "area",
  "encoding": {
    "x": {"name": "name","type": "N"},
    "y": {"name": "Miles_per_Gallon","type": "Q"},
    "color": {"name": "Cylinders","type": "O"}
  },
  "config": {"singleHeight": 400,"singleWidth": 400,"largeBandMaxCardinality": 20},
  "data": {"url": "data/cars.json"}
}
```

**Comment by kanitw** [comment]
There are a few issues here

1. the “name" variable should be title case (“Name”).  — I have make error a little clearer for a function call.  
2. There is a bug in stacking call.  I just fixed this in #541.

3. Even though now it renders correctly, it is worth noting that both of these are bad visualization designs as both encoding you generated can lead to occlusion.  In the future, we plan to integrate Compass with Polestar so we can show warnings if users specify badly designed encodings.
